### ISSUE TITLE
Doc enhancement: what format should the training data follow?

### ISSUE THREAD

**Issue by topspinj** [uber/ludwig#583]
**Is your feature request related to a problem? Please describe.**
It would be ideal if there was more documentation around how the training data should be formatted. It would also be helpful to know what packages are being used behind the scenes for sentiment analysis, NER, text classification, etc. 

**Describe the use case**
For example, I want to test out sentiment analysis. Is this built off of a pre-trained model? Would it expect labels like "positive", "negative", "neutral", or can we have as many labels as we'd like? How is this different from text classification?

**Describe the solution you'd like**
More user documentation with examples, information on what packages a model is using, and what training data it's expecting.

**Comment by msaisumanth** [comment]
please take a look at the [user guide](https://uber.github.io/ludwig/examples/?from=%40). It has examples on how to do text classification. Sentiment classification is a type of text classification.

**Comment by w4nderlust** [comment]
@topspinj there's plenty of explanations about the data, can you be more specific about what you are missing? In particular for sentiment analysis just provide a csv with one column containing the text and one contaning the sentiment class, any label for the class would work, you know what the labels mean, and no, there's no difference wrt text classification. Regarding all the tasks, there's no "package being used", all model are trained from scratch on TensorFlow.
